core/src/main/java/org/apache/accumulo/core/clientImpl/OfflineIterator.java

```diff
@@ -345,9 +245,13 @@ private SortedKeyValueIterator<Key,Value> createIterator(KeyExtent extent,
 
     MultiIterator multiIter = new MultiIterator(readers, extent);
 
-    OfflineIteratorEnvironment iterEnv =
-        new OfflineIteratorEnvironment(context, tableId, authorizations, tableCC, false,
-            samplerConfImpl == null ? null : samplerConfImpl.toSamplerConfiguration());
+    ClientIteratorEnvironment.Builder iterEnvBuilder =
+        new ClientIteratorEnvironment.Builder().withAuthorizations(authorizations)
+            .withScope(IteratorScope.scan).withTableId(tableId).withClient(context);
+    if (scannerSamplerConfig != null) {
+      iterEnvBuilder.withSamplerConfiguration(scannerSamplerConfig);
+    }
+    IteratorEnvironment iterEnv = iterEnvBuilder.build();
```

Comment:
It seems usage of `tableCC` and `samplerConfImpl` were dropped. Was this intentional?

Reply:
It also seems like `isSamplingEnabled()` is always false in original impl, but not in this impl

Reply:
The old code set useSampling to `false`, but went to the trouble of creating the `sampleConfiguration`. I'm wondering if setting `useSampling` to `false` was a bug in the old version.

https://github.com/apache/accumulo/blob/3912b36d0b0ca103b7068d7c7575c99d32a11a97/core/src/main/java/org/apache/accumulo/core/clientImpl/OfflineIterator.java#L324-L331

Reply:
Regarding the missing `tableCC`, by setting `withClient(ClientContext)`, it will create a `ClientServiceEnvironmentImpl` for the calls to ` getServiceEnv` and `getPluginEnv` and the table configuration can be retrieved from that object.

Reply:
Fixed the `samplerConfImpl` issue in c14c35f

Reply:
Fixed in 3cec5f7